### PR TITLE
types/netmap, all: use read-only tailcfg.NodeView in NetworkMap

### PR DESCRIPTION
**Pull request by bradfitz**
Updates #8948

### CODE REVIEW

File: ipn/ipnlocal/local.go [hunk 1]
```diff
@@ -2438,36 +2442,37 @@ func (b *LocalBackend) Ping(ctx context.Context, ip netip.Addr, pingType tailcfg
 	}
 }
 
-func (b *LocalBackend) pingPeerAPI(ctx context.Context, ip netip.Addr) (peer *tailcfg.Node, peerBase string, err error) {
+func (b *LocalBackend) pingPeerAPI(ctx context.Context, ip netip.Addr) (peer tailcfg.NodeView, peerBase string, err error) {
+	var zero tailcfg.NodeView
```
Comment (maisem) [inline ipn/ipnlocal/local.go:2446]: I've been calling this noNode/noUser in a few places.

File: net/tsdial/dnsmap_test.go [hunk 2]
```diff
@@ -12,6 +12,14 @@ import (
 	"tailscale.com/types/netmap"
 )
 
+func nodeViews(v []*tailcfg.Node) []tailcfg.NodeView {
+	nv := make([]tailcfg.NodeView, len(v))
```
Comment (maisem) [inline net/tsdial/dnsmap_test.go:16]: Could also do
views.SliceOfViews(v).AsSlice()

Comment (maisem) [reply net/tsdial/dnsmap_test.go:16]: Considering it's repeated a few times

**Review by maisem — APPROVED** [review]

File: ipn/ipnlocal/local.go (hunk 1 above)
Comment (bradfitz) [reply ipn/ipnlocal/local.go:2446]: I'm getting ready for https://github.com/golang/go/issues/61372 in which cas we just need to delete this declaration line and it'll keep working.

File: net/tsdial/dnsmap_test.go (hunk 2 above)
Comment (bradfitz) [reply net/tsdial/dnsmap_test.go:16]: I'm too dumb for that.

I can't follow these docs:

```go
// SliceOfViews returns a ViewSlice for x.
func SliceOfViews[T ViewCloner[T, V], V StructView[T]](x []T) SliceView[T, V] {
	return SliceView[T, V]{x}
}

// SliceView is a read-only wrapper around a struct which should only be exposed
// as a View.
type SliceView[T ViewCloner[T, V], V StructView[T]] struct {
```

"a struct which should only be exposed as a View".

I have no structs here?

**Pull request merged**